cmd/kubeadm/app/util/version.go

```diff
@@ -24,17 +24,20 @@ import (
 	"strings"
 	"time"
 
+	"github.com/golang/glog"
 	netutil "k8s.io/apimachinery/pkg/util/net"
+	versionutil "k8s.io/kubernetes/pkg/util/version"
+	"k8s.io/kubernetes/pkg/version"
 )
 
 const (
-	getReleaseVersionTimeout = time.Duration(10 * time.Second)
+	getReleaseVersionTimeout = time.Duration(4 * time.Second)
```

Comment:
10 seconds are actually good enough for cases with broken proxy+firewalls, which are seen as "connection timeout".
4 seconds can be too low in case of slow hardware (raspberry) + remote location (Australia, China).
In my test environment, even on good HW and quite big channels, from China it takes about 3 seconds for `curl` to get first stage HTTP reply. It is needed to be kept in mind, that `dl.k8s.io` first return redirect (HTTP 302), and then there is second "behind the scenes" http call to `https://storage.googleapis.com/kubernetes-release/....` to get actual content of label file.

Reply:
4 might be too low, indeed.
10 seems fine but the problem is that air-gapped users would have to wait "a bit" on **each** kubeadm command.
i will change it back.

Reply:
This timeout only will be hit in the scenarios where air gaped environment has firewall that silently drop packets. In that case, yes, it will be 10 seconds waiting period. In the scenarios where firewall is actively denying connections or there is no route from the environment to the internet, after receiving ICMP "unreachable" packets, http client will immediately produce error.

Reply:
makes sense. will change it back to 10 after we clear the discussion.

i was also browsing some bash scripts today (forgot where in k8s) and i saw some code that does a `curl` on the redirected address like:
`https://storage.googleapis.com/kubernetes-release/release/latest-1.11.txt`
instead of:
`https://dl.k8s.io/release/latest-1.11.txt`

do you think we can do that to help slow connections?